ethpm_cli/manifest.py

```diff
@@ -0,0 +1,193 @@
+    cli_logger.info("Follow the steps in the docs to generate solc output.")
+
+    contracts_dir = project_dir / "contracts"
+
+    validate_solc_output(project_dir)
+    solc_output_path = project_dir / SOLC_OUTPUT
+    solc_output = json.loads(solc_output_path.read_text())["contracts"]
+
+    builder_fns = (
+        gen_package_name(),
+        gen_version(),
+        gen_manifest_version(),
+        gen_description(),
+        gen_license(),
+        gen_authors(),
+        gen_keywords(),
+        gen_links(),
+        *gen_contract_types(solc_output),
+        *gen_sources(solc_output, contracts_dir),
+        # todo: *gen_build_dependencies(),
+        # todo: *gen_deployments,
+        # todo: ipfs pinning support
+        gen_validate_manifest(),
+        b.write_to_disk(project_dir),
+    )
+    final_fns = (fn for fn in builder_fns if fn is not None)
+    manifest = b.build({}, *final_fns)
+    cli_logger.info(
+        f"Manifest successfully created and written to {project_dir}/{manifest['version']}.json."
+    )
+
+
+def gen_validate_manifest() -> Optional[Callable[..., Manifest]]:
+    flag = parse_bool_flag(
+        "Would you like to validate your manifest against the json schema? (recommended) (y/n) "
+    )
+    if flag:
+        return b.validate()
+    return None
+
+
+def gen_contract_types(
+    solc_output: Dict[str, Any]
+) -> Iterable[Callable[..., Manifest]]:
+    contract_types = get_contract_types(solc_output)
+    pretty = "\n".join(contract_types)
+    flag = parse_bool_flag(
+        f"{len(contract_types)} contract types available.\n\n"
+        f"{pretty}. \n"
+        "Would you like to include all available contract types? (y/n)\n"
+    )
+    if not flag:
+        raise Exception("Custom contract types are not supported yet.")
+    return build_contract_types(contract_types, solc_output)
+
+
+def gen_sources(
+    solc_output: Dict[str, Any], contracts_dir: Path
+) -> Iterable[Callable[..., Manifest]]:
+    available_sources = [
+        str(src.relative_to(contracts_dir)) for src in contracts_dir.glob("**/*.sol")
+    ]
+    contract_types = get_contract_types(solc_output)
+    pretty = "\n".join(sorted(available_sources))
+    flag = parse_bool_flag(
+        f"{len(available_sources)} sources available.\n\n"
+        f"{pretty}. \n"
+        "Would you like to include all available sources? (y/n)\n"
+    )
+    if not flag:
+        raise Exception("sorry, we dont support specific source selection yet.")
+    inline_source_flag = parse_bool_flag(
+        "Would you like to automatically inline all sources? (y/n) "
+    )
+    if not inline_source_flag:
+        raise Exception("Sorry, we dont support pinning sources yet.")
+    # todo: validate a source for every contract type
+    return build_sources(contract_types, solc_output, contracts_dir)
+
+
+def gen_package_name() -> Callable[..., Manifest]:
+    package_name = input("Enter your package's name: ")
+    return b.package_name(package_name)
```

Comment:
Does this validate the name?  If not, you might validate the name before moving on and throw a friendly error at this stage.